Transient IT failure for test_local_chalice_index_endpoints fails with 500

> https://gitlab.explore.anvilproject.org/ucsc/azul/-/jobs/55488/raw 

 _Originally posted by @achave11-ucsc in [#4784](https://github.com/DataBiosphere/azul/issues/4784#issuecomment-3474991483)_

I originally thought this was a different flavor of #4784, but it may not be the case and I also omitted evidence in original post.

This happened for `hammerbox` during the promotion branch build, these changes included required reindexing. The retried pipeline succeeded.

```
test_local_chalice_index_endpoints (integration_test.AzulChaliceLocalIntegrationTest.test_local_chalice_index_endpoints) ... 2025-10-31 21:30:56,634    INFO Thread-4 (process_request_thread) azul.chalice: Received GET request for '/index/files', with {"query": {"catalog": "anvil11-it"}, "headers": {"host": "127.0.0.1:8000", "user-agent": "python-requests/2.32.5", "accept-encoding": "gzip, deflate", "accept": "*/*", "connection": "keep-alive"}}.
2025-10-31 21:30:56,635    INFO Thread-4 (process_request_thread) azul.chalice: … without a request body
2025-10-31 21:30:56,635    INFO Thread-4 (process_request_thread) azul.chalice: Did not authenticate request.
2025-10-31 21:30:56,660    INFO Thread-4 (process_request_thread) botocore.credentials: Found credentials from IAM Role: azul-gitlab
2025-10-31 21:30:56,784    INFO Thread-4 (process_request_thread) azul.deployment: Allocated new Boto3 client for 'dynamodb' with ID 140636835460304
2025-10-31 21:30:56,787    INFO Thread-4 (process_request_thread) azul.boto3: dynamodb.GetItem:	Making POST request to https://dynamodb.us-east-1.amazonaws.com/
2025-10-31 21:30:56,787    INFO Thread-4 (process_request_thread) azul.boto3: dynamodb.GetItem:	… with a request body of length 142 being b'{"TableName": "azul-sources_cache_by_auth-hammerbox", "Key": {"identity": {"S": "anvil11-it:"}}, "ProjectionExpression": "sources,expiration"}'
2025-10-31 21:30:56,834    INFO Thread-4 (process_request_thread) azul.boto3: dynamodb.GetItem:	Got 200 response
2025-10-31 21:30:56,834    INFO Thread-4 (process_request_thread) azul.boto3: dynamodb.GetItem:	… with a response body of length 2 being b'{}'
2025-10-31 21:30:56,848    INFO Thread-4 (process_request_thread) azul.deployment: Allocated new Boto3 client for 'secretsmanager' with ID 140636803648208
2025-10-31 21:30:57,083    INFO Thread-4 (process_request_thread) azul.terra: Making GET request to 'https://data.terra.bio/api/repository/v1/snapshots/roleMap'
2025-10-31 21:30:57,083    INFO Thread-4 (process_request_thread) azul.terra: … without a request body
2025-10-31 21:30:57,084    INFO Thread-4 (process_request_thread) azul.terra: … with request headers [('Connection', 'close'), ('x-goog-api-client', 'cred-type/sa'), ('authorization', 'REDACTED')]
2025-10-31 21:30:57,647    INFO Thread-4 (process_request_thread) azul.terra: Got 200 response after 0.563s from GET to https://data.terra.bio/api/repository/v1/snapshots/roleMap
2025-10-31 21:30:57,647    INFO Thread-4 (process_request_thread) azul.terra: … with response headers HTTPHeaderDict({'Date': 'Fri, 31 Oct 2025 21:30:57 GMT', 'Server': 'Apache', 'X-Frame-Options': 'SAMEORIGIN', 'Access-Control-Allow-Headers': 'DNT,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Range,Authorization,Accept,Referer,X-App-Id,Origin', 'Access-Control-Allow-Methods': 'GET,POST,DELETE,PUT,PATCH,OPTIONS,HEAD', 'X-Content-Type-Options': 'nosniff', 'Strict-Transport-Security': 'max-age=31536000;includeSubDomains', 'Cache-Control': 'no-cache,no-store,must-revalidate', 'X-Request-ID': '6gbjyEz3', 'Content-Type': 'application/json', 'Content-Length': '145750', 'Set-Cookie': 'JSESSIONID=DF059803918F22C2C9F45A1CBA996D71; Path=/; Secure; HttpOnly', 'Vary': 'Accept-Encoding,Origin', 'Via': '1.1 google', 'Alt-Svc': 'h3=":443"; ma=2592000,h3-29=":443"; ma=2592000', 'Connection': 'close'})
2025-10-31 21:30:57,647    INFO Thread-4 (process_request_thread) azul.terra: … with a response body of length 145750 starting in b'{"roleMap":{"81e4a4c1-2341-46f6-9af1-005097500086":["discoverer","reader","aggregate_data_reader"],"c5aeee20-d383-4c09-8a78-a7c810af7857":["discoverer","reader","aggregate_data_reader"],"82336861-0e39-497f-bdbe-03923f5f36ab":["discoverer","reader","aggregate_data_reader"],"4a94d70f-5335-410c-a0b0-909226481259":["discoverer","reader","aggregate_data_reader"],"bc7f344d-ff18-4a10-8f59-518324ad4126":["discoverer","reader","aggregate_data_reader"],"c25f206f-c5de-48f2-9dfa-7ab41221d6eb":["discoverer","reader","aggregate_data_reader"],"bbe3aa2e-42b0-44db-83c8-a4b9ec8c31ab":["discoverer","reader","aggregate_data_reader"],"96bd5e38-1f50-48d8-a78b-00d89a1ad105":["discoverer","reader","aggregate_data_reader"],"0cadeadc-7470-4e57-8de8-a3e320c3e6a8":["discoverer","reader","aggregate_data_reader"],"5397053b-33a7-456d-9b2c-66c65483734e":["discoverer","reader","aggregate_data_reader"],"a17642d0-5b55-407d-b205-e7a849a49bc7":["discoverer","reader","aggregate_data_reader"],"a0299a62-22f3-497c-bacb-53f1a2d9b387":["discoverer"...'
2025-10-31 21:30:57,655    INFO Thread-4 (process_request_thread) azul.boto3: dynamodb.PutItem:	Making POST request to https://dynamodb.us-east-1.amazonaws.com/
2025-10-31 21:30:57,655    INFO Thread-4 (process_request_thread) azul.boto3: dynamodb.PutItem:	… with a request body of length 70504 starting in b'{"TableName": "azul-sources_cache_by_auth-hammerbox", "Item": {"identity": {"S": "anvil11-it:"}, "sources": {"S": "[\\"6e091976-9ee3-4141-a147-ef0851272dd5\\", \\"9e053aaa-b55e-41fe-84ac-77045891b3b3\\", \\"ff2e5d3c-e8d2-4cea-953f-9a53d244b098\\", \\"987caf73-de2c-4f90-92a8-d50fc3052d32\\", \\"6a9b080f-00ec-4673-9e5c-77b251233851\\", \\"457bd7a8-be1c-4e28-86aa-39f3fad2112c\\", \\"c95bf55d-7f4e-47b0-af97-198823f21aaa\\", \\"e5336126-9e09-4bc6-a9ef-23de0dfc975f\\", \\"a474ad8c-4440-4eb2-8e26-2ff407e04d9e\\", \\"84d21e92-a757-482f-8189-b53324310494\\", \\"ffa21f49-0fb8-4192-a7ed-9cd884e81c6d\\", \\"d6b24a04-73b8-4273-a349-954e0ce57236\\", \\"b9bfea91-8aa2-4e25-8d3b-86cedb3674b7\\", \\"113b5a39-4d59-4b88-bfe1-aeef31fd0b21\\", \\"1542673a-aeda-48a4-81c5-77c7a8f763af\\", \\"160fd608-d1d4-40d8-929b-202d721f4e4b\\", \\"37d70dc7-a02b-49df-b0ae-4cf51842d0ea\\", \\"d85d5155-b47c-4001-82a5-82ee8931a7f1\\", \\"9df9d3c5-44c7-4d5e-9e5b-0f703f69f3e0\\", \\"88d0385d-63fd-40cf-bcae-5eb530e65fc8\\", \\"5401f09c-1fb4-4d9c-92a3-8bb61fdd5bf8\\", \\"16db2325-49f2-4e33-b5...'
2025-10-31 21:30:57,708    INFO Thread-4 (process_request_thread) azul.boto3: dynamodb.PutItem:	Got 200 response
2025-10-31 21:30:57,708    INFO Thread-4 (process_request_thread) azul.boto3: dynamodb.PutItem:	… with a response body of length 2 being b'{}'
2025-10-31 21:30:57,757    INFO Thread-4 (process_request_thread) azul.deployment: Allocated new Boto3 client for 'es' with ID 140636803657616
2025-10-31 21:30:57,929    INFO Thread-4 (process_request_thread) opensearch: Making POST request to https://vpc-azul-index-anvilprod-ggipah4skn2ftt47u4xgvydzqm.us-east-1.es.amazonaws.com:443/azul_v2_hammerbox_anvil11-it_files_aggregate/_search
2025-10-31 21:30:57,929    INFO Thread-4 (process_request_thread) opensearch: … with a request body of length 1379226 starting in b'{"post_filter":{"bool":{"must":[{"constant_score":{"filter":{"terms":{"sources.id.keyword":["004ee0a4-7672-4922-9ec2-9f1c4e1ff4a9","0070f6a4-5430-47aa-b1fb-f7d175a470c8","009e77b1-362e-47e0-9ae7-a03a390a07ce","00c1a1f1-6bfc-478b-8b32-5d5911081638","00d1422e-2055-4a74-af2e-c7ab1e25cd5e","01795dd6-5123-4ed4-88de-06a1a99f75af","01a8a72d-0a21-4bbf-b50f-6f1687cf1e61","01d0e707-2021-4740-9b21-60c8ad24fc76","01e26559-f5a5-42f8-b601-c24ef42f4aba","01ebc9c6-08de-41e5-a689-aaa5685a1d92","020e1f1a-7314-44b5-a281-942a45269815","02137a6c-e68a-4ccb-b5aa-e1013d8e44da","023486a4-2a09-4a19-aba4-f717a7483e6e","02378204-6234-413a-8905-95cf2454de93","026ea946-27df-403e-8215-511f1cdf2537","029f160a-b25f-44cd-ad7e-98ee3709135b","02a58bdd-4afb-4d36-ae6d-ebe8b1f3f951","02bb75f3-e9c1-4ae4-bf99-ec8115cb1adf","02cb6c0a-5048-45ad-99aa-d03602f8453a","03095f7b-2131-4885-a76e-41853bc8ec54","032281c4-1f6b-47fb-a7ac-b3e0871eeadc","03409b70-6310-402a-a264-9147ea43ae9b","03447627-4e56-4ec5-a0c5-01b7eb71201c","0345a707-9b97-4d03-964a-7366809...'
2025-10-31 21:30:58,026    INFO Thread-4 (process_request_thread) opensearch: Got 200 response after 0.097s from POST to https://vpc-azul-index-anvilprod-ggipah4skn2ftt47u4xgvydzqm.us-east-1.es.amazonaws.com:443/azul_v2_hammerbox_anvil11-it_files_aggregate/_search
2025-10-31 21:30:58,026    INFO Thread-4 (process_request_thread) opensearch: … with a response body of length 23615 starting in '{"took":63,"timed_out":false,"_shards":{"total":1,"successful":1,"skipped":0,"failed":0},"hits":{"total":{"value":1569,"relation":"eq"},"max_score":null,"hits":[{"_index":"azul_v2_hammerbox_anvil11-it_files_aggregate","_id":"239a5386-318d-415b-bf47-2b6834dc1e3e","_score":null,"_source":{"entity_id":"239a5386-318d-415b-bf47-2b6834dc1e3e","contents":{"activities":[],"biosamples":[],"datasets":[{"document_id":["e5aee011-bdb3-4caa-954c-a46678656270"],"source_datarepo_row_ids":["workspace_attributes:5b74f9e4-4d23-42de-97e1-a735ac1a27fc"],"dataset_id":["c84917f5-df6d-6be2-94fe-9e178839220e"],"consent_group":["NRES"],"data_use_permission":["NRES"],"owner":["Kristin Ardlie"],"principal_investigator":["~null"],"registered_identifier":["none"],"title":["ANVIL_GTEx_public_data"],"data_modality":["genomic","transcriptomic"]}],"diagnoses":[],"donors":[],"files":[{"document_id":"239a5386-318d-415b-bf47-2b6834dc1e3e","source_datarepo_row_ids":["file_inventory:224be816-6186-4a3f-8100-a794e560de24"],"file_id":"0006fb57-ce17-…'
2025-10-31 21:30:58,030   ERROR Thread-4 (process_request_thread) azul-service-hammerbox: Caught exception for path /index/{entity_type}
Traceback (most recent call last):
  File "/build/.venv/lib/python3.13/site-packages/chalice/app.py", line 1922, in _get_view_function_response
    response = view_function(**function_args)
  File "/builds/ucsc/azul/lambdas/service/app.py", line 953, in repository_search
    return app.repository_controller.search(catalog=app.catalog,
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^
                                            entity_type=entity_type,
                                            ^^^^^^^^^^^^^^^^^^^^^^^^
    ...<2 lines>...
                                            pagination=app.get_pagination(entity_type),
                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                                            authentication=request.authentication)
                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/builds/ucsc/azul/src/azul/service/repository_controller.py", line 109, in search
    response = self.service.search(catalog=catalog,
                                   entity_type=entity_type,
    ...<2 lines>...
                                   filters=filters,
                                   pagination=pagination)
  File "/builds/ucsc/azul/src/azul/service/repository_service.py", line 136, in search
    response = self._search(catalog=catalog,
                            filters=filters,
    ...<2 lines>...
                            entity_type=entity_type,
                            file_url_func=file_url_func)
  File "/builds/ucsc/azul/src/azul/service/repository_service.py", line 226, in _search
    response = chain.process_response(response)
  File "/builds/ucsc/azul/src/azul/service/elasticsearch_service.py", line 141, in process_response
    response2: R2 = self.outer.process_response(response1)
                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^
  File "/builds/ucsc/azul/src/azul/plugins/metadata/anvil/service/response.py", line 109, in process_response
    hits=list(map(self._make_hit, hits)),
         ~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/builds/ucsc/azul/src/azul/plugins/metadata/anvil/service/response.py", line 180, in _make_hit
    'sources': list(map(self._make_source, json_element_mappings(sources))),
               ~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/builds/ucsc/azul/src/azul/plugins/metadata/anvil/service/response.py", line 187, in _make_source
    self._special_fields.source_prefix: json_str(es_source['prefix']),
                                                 ~~~~~~~~~^^^^^^^^^^
KeyError: 'prefix'
2025-10-31 21:30:58,036    INFO Thread-4 (process_request_thread) azul.chalice: Returning 500 response with headers {"headers": {"Access-Control-Allow-Origin": "*", "Access-Control-Allow-Headers": "Authorization,Content-Type,X-Amz-Date,X-Amz-Security-Token,X-Api-Key", "Content-Security-Policy": "default-src 'self';img-src 'self' data:;script-src 'self';style-src 'self';frame-ancestors 'none';form-action 'self'", "Referrer-Policy": "strict-origin-when-cross-origin", "Strict-Transport-Security": "max-age=63072000; includeSubDomains; preload", "X-Content-Type-Options": "nosniff", "X-Frame-Options": "DENY", "X-XSS-Protection": "1; mode=block", "Cache-Control": "no-store"}}.
2025-10-31 21:30:58,036    INFO Thread-4 (process_request_thread) azul.chalice: … with a response body of length 133 being {"Code": "InternalServerError", "Message": "An internal server error occurred.", "RequestId": "39372e61-8919-4b5e-a107-3483a25636ae"}
127.0.0.1 - - [31/Oct/2025 21:30:58] "GET /index/files?catalog=anvil11-it HTTP/1.1" 500 -
FAIL

…

======================================================================
FAIL: test_local_chalice_index_endpoints (integration_test.AzulChaliceLocalIntegrationTest.test_local_chalice_index_endpoints)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builds/ucsc/azul/test/integration_test.py", line 1806, in test_local_chalice_index_endpoints
    self.assertEqual(200, response.status_code, response.content)
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: 200 != 500 : b'{"Code":"InternalServerError","Message":"An internal server error occurred.","RequestId":"39372e61-8919-4b5e-a107-3483a25636ae"}'

----------------------------------------------------------------------
Ran 16 tests in 516.699s

FAILED (failures=1)
make: *** [Makefile:259: integration_test] Error 1

```

Happened again, but in `prod` build during IT, https://gitlab.azul.data.humancellatlas.org/ucsc/azul/-/jobs/86768

```
test_local_chalice_index_endpoints (integration_test.AzulChaliceLocalIntegrationTest.test_local_chalice_index_endpoints) ... 2025-11-01 15:48:31,783    INFO Thread-4 (process_request_thread) azul.chalice: Received GET request for '/index/files', with {"query": {"catalog": "dcp54-it"}, "headers": {"host": "127.0.0.1:8000", "user-agent": "python-requests/2.32.5", "accept-encoding": "gzip, deflate", "accept": "*/*", "connection": "keep-alive"}}.
2025-11-01 15:48:31,784    INFO Thread-4 (process_request_thread) azul.chalice: … without a request body
2025-11-01 15:48:31,784    INFO Thread-4 (process_request_thread) azul.chalice: Did not authenticate request.
2025-11-01 15:48:31,838    INFO Thread-4 (process_request_thread) botocore.credentials: Found credentials from IAM Role: azul-gitlab
2025-11-01 15:48:32,049    INFO Thread-4 (process_request_thread) azul.deployment: Allocated new Boto3 client for 'dynamodb' with ID 140460621644528
2025-11-01 15:48:32,053    INFO Thread-4 (process_request_thread) azul.boto3: dynamodb.GetItem:	Making POST request to https://dynamodb.us-east-1.amazonaws.com/
2025-11-01 15:48:32,054    INFO Thread-4 (process_request_thread) azul.boto3: dynamodb.GetItem:	… with a request body of length 135 being b'{"TableName": "azul-sources_cache_by_auth-prod", "Key": {"identity": {"S": "dcp54-it:"}}, "ProjectionExpression": "sources,expiration"}'
2025-11-01 15:48:32,123    INFO Thread-4 (process_request_thread) azul.boto3: dynamodb.GetItem:	Got 200 response
2025-11-01 15:48:32,123    INFO Thread-4 (process_request_thread) azul.boto3: dynamodb.GetItem:	… with a response body of length 2 being b'{}'
2025-11-01 15:48:32,164    INFO Thread-4 (process_request_thread) azul.deployment: Allocated new Boto3 client for 'secretsmanager' with ID 140460621647216
2025-11-01 15:48:32,510    INFO Thread-4 (process_request_thread) azul.terra: Making GET request to 'https://data.terra.bio/api/repository/v1/snapshots/roleMap'
2025-11-01 15:48:32,511    INFO Thread-4 (process_request_thread) azul.terra: … without a request body
2025-11-01 15:48:32,512    INFO Thread-4 (process_request_thread) azul.terra: … with request headers [('Connection', 'close'), ('x-goog-api-client', 'cred-type/sa'), ('authorization', 'REDACTED')]
2025-11-01 15:48:33,175    INFO Thread-4 (process_request_thread) azul.terra: Got 200 response after 0.663s from GET to https://data.terra.bio/api/repository/v1/snapshots/roleMap
2025-11-01 15:48:33,175    INFO Thread-4 (process_request_thread) azul.terra: … with response headers HTTPHeaderDict({'Date': 'Sat, 01 Nov 2025 15:48:33 GMT', 'Server': 'Apache', 'X-Frame-Options': 'SAMEORIGIN', 'Access-Control-Allow-Headers': 'DNT,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Range,Authorization,Accept,Referer,X-App-Id,Origin', 'Access-Control-Allow-Methods': 'GET,POST,DELETE,PUT,PATCH,OPTIONS,HEAD', 'X-Content-Type-Options': 'nosniff', 'Strict-Transport-Security': 'max-age=31536000;includeSubDomains', 'Cache-Control': 'no-cache,no-store,must-revalidate', 'X-Request-ID': 'eyLwJN4z', 'Content-Type': 'application/json', 'Content-Length': '145750', 'Set-Cookie': 'JSESSIONID=AA08666D2924CC00C7BF5723C872BFEF; Path=/; Secure; HttpOnly', 'Vary': 'Accept-Encoding,Origin', 'Via': '1.1 google', 'Alt-Svc': 'h3=":443"; ma=2592000,h3-29=":443"; ma=2592000', 'Connection': 'close'})
2025-11-01 15:48:33,176    INFO Thread-4 (process_request_thread) azul.terra: … with a response body of length 145750 starting in b'{"roleMap":{"81e4a4c1-2341-46f6-9af1-005097500086":["discoverer","reader","aggregate_data_reader"],"c5aeee20-d383-4c09-8a78-a7c810af7857":["discoverer","reader","aggregate_data_reader"],"82336861-0e39-497f-bdbe-03923f5f36ab":["discoverer","reader","aggregate_data_reader"],"4a94d70f-5335-410c-a0b0-909226481259":["discoverer","reader","aggregate_data_reader"],"bc7f344d-ff18-4a10-8f59-518324ad4126":["discoverer","reader","aggregate_data_reader"],"c25f206f-c5de-48f2-9dfa-7ab41221d6eb":["discoverer","reader","aggregate_data_reader"],"bbe3aa2e-42b0-44db-83c8-a4b9ec8c31ab":["discoverer","reader","aggregate_data_reader"],"96bd5e38-1f50-48d8-a78b-00d89a1ad105":["discoverer","reader","aggregate_data_reader"],"0cadeadc-7470-4e57-8de8-a3e320c3e6a8":["discoverer","reader","aggregate_data_reader"],"5397053b-33a7-456d-9b2c-66c65483734e":["discoverer","reader","aggregate_data_reader"],"a17642d0-5b55-407d-b205-e7a849a49bc7":["discoverer","reader","aggregate_data_reader"],"a0299a62-22f3-497c-bacb-53f1a2d9b387":["discoverer"...'
2025-11-01 15:48:33,185    INFO Thread-4 (process_request_thread) azul.boto3: dynamodb.PutItem:	Making POST request to https://dynamodb.us-east-1.amazonaws.com/
2025-11-01 15:48:33,185    INFO Thread-4 (process_request_thread) azul.boto3: dynamodb.PutItem:	… with a request body of length 70497 starting in b'{"TableName": "azul-sources_cache_by_auth-prod", "Item": {"identity": {"S": "dcp54-it:"}, "sources": {"S": "[\\"d1c71bed-db80-4a01-a9b8-d5422db8a1f0\\", \\"7643b788-4c38-4ea2-aefe-9ceec500b238\\", \\"22e0626e-869c-4564-9d01-4923528d6ebb\\", \\"b73d3736-6abe-4180-b5ae-d7a69249d545\\", \\"391e376f-6f5b-4ea5-aa3a-92748403ce85\\", \\"a395c817-826a-41c5-a692-d52352b7a5a1\\", \\"8c6314aa-ff6a-4198-9f9e-eff7aa029ca9\\", \\"f3283538-c86e-4d2b-b6fb-c51d3479b9cc\\", \\"763be033-5065-4511-aa7a-9172d82b3ee0\\", \\"f52e9f35-cbbb-43f9-bb87-94f9d2788bf6\\", \\"166bb981-cd83-46df-9b44-5fe20d935fe0\\", \\"faf1d08b-b176-4a94-a6eb-869f0af5038b\\", \\"61dd201a-d4ab-42df-8f06-dd905877ba68\\", \\"c8f4520b-74b9-44ed-b2e0-140a25aab5ba\\", \\"a299ab9e-92c5-4ca8-b110-2000d7d3e4a7\\", \\"3ccf79bc-89b8-4448-96e9-605e05d0e7b4\\", \\"86e20ce0-95f1-40e7-8888-1bf9be974f6d\\", \\"cfb2b9ff-0fda-42cd-a565-cfea5f2099c2\\", \\"5c0647c0-72a0-4216-8b83-ea22cf96a4e2\\", \\"1a859ad4-dc77-4d64-b698-a602a5c8e051\\", \\"97899c5b-ad88-4b59-a878-d0c63c37fdbc\\", \\"b0571deb-35bd-49d3-84ad-e298...'
2025-11-01 15:48:33,221    INFO Thread-4 (process_request_thread) azul.boto3: dynamodb.PutItem:	Got 200 response
2025-11-01 15:48:33,221    INFO Thread-4 (process_request_thread) azul.boto3: dynamodb.PutItem:	… with a response body of length 2 being b'{}'
2025-11-01 15:48:33,286    INFO Thread-4 (process_request_thread) azul.deployment: Allocated new Boto3 client for 'es' with ID 140460621656624
2025-11-01 15:48:33,511    INFO Thread-4 (process_request_thread) opensearch: Making POST request to https://vpc-azul-index-prod-mjluewlvxikvhillefxotw2tbm.us-east-1.es.amazonaws.com:443/azul_v2_prod_dcp54-it_files_aggregate/_search
2025-11-01 15:48:33,511    INFO Thread-4 (process_request_thread) opensearch: … with a request body of length 2627548 starting in b'{"post_filter":{"bool":{"must":[{"constant_score":{"filter":{"terms":{"sources.id.keyword":["004ee0a4-7672-4922-9ec2-9f1c4e1ff4a9","0070f6a4-5430-47aa-b1fb-f7d175a470c8","009e77b1-362e-47e0-9ae7-a03a390a07ce","00c1a1f1-6bfc-478b-8b32-5d5911081638","00d1422e-2055-4a74-af2e-c7ab1e25cd5e","01795dd6-5123-4ed4-88de-06a1a99f75af","01a8a72d-0a21-4bbf-b50f-6f1687cf1e61","01d0e707-2021-4740-9b21-60c8ad24fc76","01e26559-f5a5-42f8-b601-c24ef42f4aba","01ebc9c6-08de-41e5-a689-aaa5685a1d92","020e1f1a-7314-44b5-a281-942a45269815","02137a6c-e68a-4ccb-b5aa-e1013d8e44da","023486a4-2a09-4a19-aba4-f717a7483e6e","02378204-6234-413a-8905-95cf2454de93","026ea946-27df-403e-8215-511f1cdf2537","029f160a-b25f-44cd-ad7e-98ee3709135b","02a58bdd-4afb-4d36-ae6d-ebe8b1f3f951","02bb75f3-e9c1-4ae4-bf99-ec8115cb1adf","02cb6c0a-5048-45ad-99aa-d03602f8453a","03095f7b-2131-4885-a76e-41853bc8ec54","032281c4-1f6b-47fb-a7ac-b3e0871eeadc","03409b70-6310-402a-a264-9147ea43ae9b","03447627-4e56-4ec5-a0c5-01b7eb71201c","0345a707-9b97-4d03-964a-7366809...'
2025-11-01 15:48:33,653    INFO Thread-4 (process_request_thread) opensearch: Got 200 response after 0.141s from POST to https://vpc-azul-index-prod-mjluewlvxikvhillefxotw2tbm.us-east-1.es.amazonaws.com:443/azul_v2_prod_dcp54-it_files_aggregate/_search
2025-11-01 15:48:33,653    INFO Thread-4 (process_request_thread) opensearch: … with a response body of length 86913 starting in '{"took":92,"timed_out":false,"_shards":{"total":1,"successful":1,"skipped":0,"failed":0},"hits":{"total":{"value":68,"relation":"eq"},"max_score":null,"hits":[{"_index":"azul_v2_prod_dcp54-it_files_aggregate","_id":"223ffd88-801e-4fdb-9cd7-e78bd05dd88f","_score":null,"_source":{"entity_id":"223ffd88-801e-4fdb-9cd7-e78bd05dd88f","contents":{"sample_specimens":[{"document_id":["1b9ab6b2-8391-4050-97ec-4da49b799d89"],"biomaterial_id":["3_8_M_Brain_Myeloid_Cortex"],"has_input_biomaterial":["~null"],"_source":["specimen_from_organism"],"disease":["~null"],"organ":["brain"],"organ_part":["~null"],"storage_method":["~null"],"preservation_method":["~null"],"_type":["specimen"]}],"samples":[{"document_id":["1b9ab6b2-8391-4050-97ec-4da49b799d89"],"biomaterial_id":["3_8_M_Brain_Myeloid_Cortex"],"entity_type":["specimens"],"organ":["brain"],"organ_part":["~null"],"model_organ":["~null"],"model_organ_part":["~null"],"effective_organ":["brain"]}],"sequencing_inputs":[{"document_id":["7e682fe2-284d-4926-bf37-5cf75b47bc5f"]…'
2025-11-01 15:48:33,663   ERROR Thread-4 (process_request_thread) azul-service-prod: Caught exception for path /index/{entity_type}
Traceback (most recent call last):
  File "/build/.venv/lib/python3.13/site-packages/chalice/app.py", line 1922, in _get_view_function_response
    response = view_function(**function_args)
  File "/builds/ucsc/azul/lambdas/service/app.py", line 953, in repository_search
    return app.repository_controller.search(catalog=app.catalog,
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^
                                            entity_type=entity_type,
                                            ^^^^^^^^^^^^^^^^^^^^^^^^
    ...<2 lines>...
                                            pagination=app.get_pagination(entity_type),
                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                                            authentication=request.authentication)
                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/builds/ucsc/azul/src/azul/service/repository_controller.py", line 109, in search
    response = self.service.search(catalog=catalog,
                                   entity_type=entity_type,
    ...<2 lines>...
                                   filters=filters,
                                   pagination=pagination)
  File "/builds/ucsc/azul/src/azul/service/repository_service.py", line 136, in search
    response = self._search(catalog=catalog,
                            filters=filters,
    ...<2 lines>...
                            entity_type=entity_type,
                            file_url_func=file_url_func)
  File "/builds/ucsc/azul/src/azul/service/repository_service.py", line 226, in _search
    response = chain.process_response(response)
  File "/builds/ucsc/azul/src/azul/service/elasticsearch_service.py", line 141, in process_response
    response2: R2 = self.outer.process_response(response1)
                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^
  File "/builds/ucsc/azul/src/azul/plugins/metadata/hca/service/response.py", line 280, in process_response
    hits=self.make_hits(hits))
         ~~~~~~~~~~~~~~^^^^^^
  File "/builds/ucsc/azul/src/azul/plugins/metadata/hca/service/response.py", line 513, in make_hits
    return list(map(self.make_hit, hits))
  File "/builds/ucsc/azul/src/azul/plugins/metadata/hca/service/response.py", line 518, in make_hit
    sources=self.make_sources(es_hit),
            ~~~~~~~~~~~~~~~~~^^^^^^^^
  File "/builds/ucsc/azul/src/azul/plugins/metadata/hca/service/response.py", line 297, in make_sources
    self._special_fields.source_prefix: s['prefix'],
                                        ~^^^^^^^^^^
KeyError: 'prefix'
2025-11-01 15:48:33,680    INFO Thread-4 (process_request_thread) azul.chalice: Returning 500 response with headers {"headers": {"Access-Control-Allow-Origin": "*", "Access-Control-Allow-Headers": "Authorization,Content-Type,X-Amz-Date,X-Amz-Security-Token,X-Api-Key", "Content-Security-Policy": "default-src 'self';img-src 'self' data:;script-src 'self';style-src 'self';frame-ancestors 'none';form-action 'self'", "Referrer-Policy": "strict-origin-when-cross-origin", "Strict-Transport-Security": "max-age=63072000; includeSubDomains; preload", "X-Content-Type-Options": "nosniff", "X-Frame-Options": "DENY", "X-XSS-Protection": "1; mode=block", "Cache-Control": "no-store"}}.
2025-11-01 15:48:33,680    INFO Thread-4 (process_request_thread) azul.chalice: … with a response body of length 133 being {"Code": "InternalServerError", "Message": "An internal server error occurred.", "RequestId": "367ff0e0-c4ae-4096-9393-867f92fd766e"}
127.0.0.1 - - [01/Nov/2025 15:48:33] "GET /index/files?catalog=dcp54-it HTTP/1.1" 500 -
FAIL
…
======================================================================
FAIL: test_local_chalice_index_endpoints (integration_test.AzulChaliceLocalIntegrationTest.test_local_chalice_index_endpoints)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builds/ucsc/azul/test/integration_test.py", line 1806, in test_local_chalice_index_endpoints
    self.assertEqual(200, response.status_code, response.content)
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: 200 != 500 : b'{"Code":"InternalServerError","Message":"An internal server error occurred.","RequestId":"367ff0e0-c4ae-4096-9393-867f92fd766e"}'

----------------------------------------------------------------------
Ran 16 tests in 650.409s

FAILED (failures=1, skipped=1)
make: *** [Makefile:259: integration_test] Error 1
```

And one more time in `anvilprod` https://gitlab.explore.anvilproject.org/ucsc/azul/-/jobs/55640:

```
test_local_chalice_index_endpoints (integration_test.AzulChaliceLocalIntegrationTest.test_local_chalice_index_endpoints) ... 2025-11-03 17:01:03,434    INFO Thread-4 (process_request_thread) azul.chalice: Received GET request for '/index/files', with {"query": {"catalog": "anvil11-it"}, "headers": {"host": "127.0.0.1:8000", "user-agent": "python-requests/2.32.5", "accept-encoding": "gzip, deflate", "accept": "*/*", "connection": "keep-alive"}}.
2025-11-03 17:01:03,435    INFO Thread-4 (process_request_thread) azul.chalice: … without a request body
2025-11-03 17:01:03,435    INFO Thread-4 (process_request_thread) azul.chalice: Did not authenticate request.
2025-11-03 17:01:03,467    INFO Thread-4 (process_request_thread) botocore.credentials: Found credentials from IAM Role: azul-gitlab
2025-11-03 17:01:03,626    INFO Thread-4 (process_request_thread) azul.deployment: Allocated new Boto3 client for 'dynamodb' with ID 140369566659792
2025-11-03 17:01:03,630    INFO Thread-4 (process_request_thread) azul.boto3: dynamodb.GetItem:	Making POST request to https://dynamodb.us-east-1.amazonaws.com/
2025-11-03 17:01:03,631    INFO Thread-4 (process_request_thread) azul.boto3: dynamodb.GetItem:	… with a request body of length 142 being b'{"TableName": "azul-sources_cache_by_auth-anvilprod", "Key": {"identity": {"S": "anvil11-it:"}}, "ProjectionExpression": "sources,expiration"}'
2025-11-03 17:01:03,682    INFO Thread-4 (process_request_thread) azul.boto3: dynamodb.GetItem:	Got 200 response
2025-11-03 17:01:03,682    INFO Thread-4 (process_request_thread) azul.boto3: dynamodb.GetItem:	… with a response body of length 2 being b'{}'
2025-11-03 17:01:03,697    INFO Thread-4 (process_request_thread) azul.deployment: Allocated new Boto3 client for 'secretsmanager' with ID 140369534847696
2025-11-03 17:01:04,003    INFO Thread-4 (process_request_thread) azul.terra: Making GET request to 'https://data.terra.bio/api/repository/v1/snapshots/roleMap'
2025-11-03 17:01:04,012    INFO Thread-4 (process_request_thread) azul.terra: … without a request body
2025-11-03 17:01:04,013    INFO Thread-4 (process_request_thread) azul.terra: … with request headers [('Connection', 'close'), ('x-goog-api-client', 'cred-type/sa'), ('authorization', 'REDACTED')]
2025-11-03 17:01:05,485    INFO Thread-4 (process_request_thread) azul.terra: Got 200 response after 1.473s from GET to https://data.terra.bio/api/repository/v1/snapshots/roleMap
2025-11-03 17:01:05,485    INFO Thread-4 (process_request_thread) azul.terra: … with response headers HTTPHeaderDict({'Date': 'Mon, 03 Nov 2025 17:01:05 GMT', 'Server': 'Apache', 'X-Frame-Options': 'SAMEORIGIN', 'Access-Control-Allow-Headers': 'DNT,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Range,Authorization,Accept,Referer,X-App-Id,Origin', 'Access-Control-Allow-Methods': 'GET,POST,DELETE,PUT,PATCH,OPTIONS,HEAD', 'X-Content-Type-Options': 'nosniff', 'Strict-Transport-Security': 'max-age=31536000;includeSubDomains', 'Cache-Control': 'no-cache,no-store,must-revalidate', 'X-Request-ID': 'ek5JO9bx', 'Content-Type': 'application/json', 'Content-Length': '145750', 'Set-Cookie': 'JSESSIONID=C91D38355617D43C2641F32A1321BE0E; Path=/; Secure; HttpOnly', 'Vary': 'Accept-Encoding,Origin', 'Via': '1.1 google', 'Alt-Svc': 'h3=":443"; ma=2592000,h3-29=":443"; ma=2592000', 'Connection': 'close'})
2025-11-03 17:01:05,486    INFO Thread-4 (process_request_thread) azul.terra: … with a response body of length 145750 starting in b'{"roleMap":{"81e4a4c1-2341-46f6-9af1-005097500086":["reader","discoverer","aggregate_data_reader"],"c5aeee20-d383-4c09-8a78-a7c810af7857":["reader","discoverer","aggregate_data_reader"],"82336861-0e39-497f-bdbe-03923f5f36ab":["reader","discoverer","aggregate_data_reader"],"4a94d70f-5335-410c-a0b0-909226481259":["reader","discoverer","aggregate_data_reader"],"bc7f344d-ff18-4a10-8f59-518324ad4126":["reader","discoverer","aggregate_data_reader"],"c25f206f-c5de-48f2-9dfa-7ab41221d6eb":["reader","discoverer","aggregate_data_reader"],"bbe3aa2e-42b0-44db-83c8-a4b9ec8c31ab":["reader","discoverer","aggregate_data_reader"],"96bd5e38-1f50-48d8-a78b-00d89a1ad105":["reader","discoverer","aggregate_data_reader"],"0cadeadc-7470-4e57-8de8-a3e320c3e6a8":["reader","discoverer","aggregate_data_reader"],"5397053b-33a7-456d-9b2c-66c65483734e":["reader","discoverer","aggregate_data_reader"],"a17642d0-5b55-407d-b205-e7a849a49bc7":["reader","discoverer","aggregate_data_reader"],"a0299a62-22f3-497c-bacb-53f1a2d9b387":["reader","di...'
2025-11-03 17:01:05,494    INFO Thread-4 (process_request_thread) azul.boto3: dynamodb.PutItem:	Making POST request to https://dynamodb.us-east-1.amazonaws.com/
2025-11-03 17:01:05,494    INFO Thread-4 (process_request_thread) azul.boto3: dynamodb.PutItem:	… with a request body of length 70504 starting in b'{"TableName": "azul-sources_cache_by_auth-anvilprod", "Item": {"identity": {"S": "anvil11-it:"}, "sources": {"S": "[\\"2e8c21bc-1392-4386-8a7a-e55834ebb910\\", \\"e489effb-bb9f-40f7-b3dc-866c5bfbf187\\", \\"1c938740-4fa8-4451-b9f4-319cf87a27ea\\", \\"22f5b413-4d03-46b2-abc6-3af9311522e9\\", \\"128f767e-2bc4-4217-9cb9-e15268610fa5\\", \\"3ab24677-3fed-42cd-b686-b82b29443feb\\", \\"3c0110cc-e0c2-43e4-bac0-3d1f389faf04\\", \\"8cfe7bb5-fa77-4f97-8169-cfdc588a93dc\\", \\"6eae34eb-d887-4794-8ba4-b63a4428b711\\", \\"747a4ee2-229e-465d-832d-477c8f712f17\\", \\"6843238c-f730-4fc9-b956-cd3526e57b37\\", \\"01d0e707-2021-4740-9b21-60c8ad24fc76\\", \\"bc364d42-c307-4bdd-992d-b94463230ad6\\", \\"2fc9cfc6-7865-41d0-a839-ea2d897429db\\", \\"b1f9255e-27d6-4ba0-befd-0de45f00a752\\", \\"44d942b9-e6a6-4ab5-9296-e151f0122ac0\\", \\"ba04e31f-ac32-42a0-8b7a-f70e5652780b\\", \\"58e35120-422d-4939-8226-e0f3e898bda6\\", \\"56c50bdc-1537-4dcc-97e0-b52e68532994\\", \\"0bf5d820-d11c-407c-a4d7-5b0aa3a4fa14\\", \\"73fa524e-7ce3-45bb-a16a-6760230cefb3\\", \\"5195603f-4fa5-4dea-8f...'
2025-11-03 17:01:05,540    INFO Thread-4 (process_request_thread) azul.boto3: dynamodb.PutItem:	Got 200 response
2025-11-03 17:01:05,540    INFO Thread-4 (process_request_thread) azul.boto3: dynamodb.PutItem:	… with a response body of length 2 being b'{}'
2025-11-03 17:01:05,595    INFO Thread-4 (process_request_thread) azul.deployment: Allocated new Boto3 client for 'es' with ID 140369534857104
2025-11-03 17:01:05,745    INFO Thread-4 (process_request_thread) opensearch: Making POST request to https://vpc-azul-index-anvilprod-ggipah4skn2ftt47u4xgvydzqm.us-east-1.es.amazonaws.com:443/azul_v2_anvilprod_anvil11-it_files_aggregate/_search
2025-11-03 17:01:05,745    INFO Thread-4 (process_request_thread) opensearch: … with a request body of length 1379226 starting in b'{"post_filter":{"bool":{"must":[{"constant_score":{"filter":{"terms":{"sources.id.keyword":["004ee0a4-7672-4922-9ec2-9f1c4e1ff4a9","0070f6a4-5430-47aa-b1fb-f7d175a470c8","009e77b1-362e-47e0-9ae7-a03a390a07ce","00c1a1f1-6bfc-478b-8b32-5d5911081638","00d1422e-2055-4a74-af2e-c7ab1e25cd5e","01795dd6-5123-4ed4-88de-06a1a99f75af","01a8a72d-0a21-4bbf-b50f-6f1687cf1e61","01d0e707-2021-4740-9b21-60c8ad24fc76","01e26559-f5a5-42f8-b601-c24ef42f4aba","01ebc9c6-08de-41e5-a689-aaa5685a1d92","020e1f1a-7314-44b5-a281-942a45269815","02137a6c-e68a-4ccb-b5aa-e1013d8e44da","023486a4-2a09-4a19-aba4-f717a7483e6e","02378204-6234-413a-8905-95cf2454de93","026ea946-27df-403e-8215-511f1cdf2537","029f160a-b25f-44cd-ad7e-98ee3709135b","02a58bdd-4afb-4d36-ae6d-ebe8b1f3f951","02bb75f3-e9c1-4ae4-bf99-ec8115cb1adf","02cb6c0a-5048-45ad-99aa-d03602f8453a","03095f7b-2131-4885-a76e-41853bc8ec54","032281c4-1f6b-47fb-a7ac-b3e0871eeadc","03409b70-6310-402a-a264-9147ea43ae9b","03447627-4e56-4ec5-a0c5-01b7eb71201c","0345a707-9b97-4d03-964a-7366809...'
2025-11-03 17:01:05,836    INFO Thread-4 (process_request_thread) opensearch: Got 200 response after 0.091s from POST to https://vpc-azul-index-anvilprod-ggipah4skn2ftt47u4xgvydzqm.us-east-1.es.amazonaws.com:443/azul_v2_anvilprod_anvil11-it_files_aggregate/_search
2025-11-03 17:01:05,836    INFO Thread-4 (process_request_thread) opensearch: … with a response body of length 24037 starting in '{"took":60,"timed_out":false,"_shards":{"total":1,"successful":1,"skipped":0,"failed":0},"hits":{"total":{"value":1141,"relation":"eq"},"max_score":null,"hits":[{"_index":"azul_v2_anvilprod_anvil11-it_files_aggregate","_id":"5adb911e-90ac-47dc-919f-5ba1b09351b2","_score":null,"_source":{"entity_id":"5adb911e-90ac-47dc-919f-5ba1b09351b2","contents":{"activities":[],"biosamples":[],"datasets":[{"document_id":["5c25d772-bdcb-4ea9-989c-f92258634ad9"],"source_datarepo_row_ids":["workspace_attributes:181145f7-239a-4426-9d54-679fe219c9e9"],"dataset_id":["1c39e90e-463c-9853-793e-7e4c8f081815"],"consent_group":["NRES"],"data_use_permission":["NRES"],"owner":["~null"],"principal_investigator":["~null"],"registered_identifier":["none"],"title":["ANVIL_T2T"],"data_modality":["assembly","genotyping","whole genome"]}],"diagnoses":[],"donors":[],"files":[{"document_id":"5adb911e-90ac-47dc-919f-5ba1b09351b2","source_datarepo_row_ids":["file_inventory:6fb2cd33-ba5d-41cd-9af8-ad6253093ab6"],"file_id":"0008461b-fd5f-33f5-8dc3-…'
2025-11-03 17:01:05,840   ERROR Thread-4 (process_request_thread) azul-service-anvilprod: Caught exception for path /index/{entity_type}
Traceback (most recent call last):
  File "/build/.venv/lib/python3.13/site-packages/chalice/app.py", line 1922, in _get_view_function_response
    response = view_function(**function_args)
  File "/builds/ucsc/azul/lambdas/service/app.py", line 953, in repository_search
    return app.repository_controller.search(catalog=app.catalog,
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^
                                            entity_type=entity_type,
                                            ^^^^^^^^^^^^^^^^^^^^^^^^
    ...<2 lines>...
                                            pagination=app.get_pagination(entity_type),
                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                                            authentication=request.authentication)
                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/builds/ucsc/azul/src/azul/service/repository_controller.py", line 109, in search
    response = self.service.search(catalog=catalog,
                                   entity_type=entity_type,
    ...<2 lines>...
                                   filters=filters,
                                   pagination=pagination)
  File "/builds/ucsc/azul/src/azul/service/repository_service.py", line 136, in search
    response = self._search(catalog=catalog,
                            filters=filters,
    ...<2 lines>...
                            entity_type=entity_type,
                            file_url_func=file_url_func)
  File "/builds/ucsc/azul/src/azul/service/repository_service.py", line 226, in _search
    response = chain.process_response(response)
  File "/builds/ucsc/azul/src/azul/service/elasticsearch_service.py", line 141, in process_response
    response2: R2 = self.outer.process_response(response1)
                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^
  File "/builds/ucsc/azul/src/azul/plugins/metadata/anvil/service/response.py", line 109, in process_response
    hits=list(map(self._make_hit, hits)),
         ~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/builds/ucsc/azul/src/azul/plugins/metadata/anvil/service/response.py", line 180, in _make_hit
    'sources': list(map(self._make_source, json_element_mappings(sources))),
               ~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/builds/ucsc/azul/src/azul/plugins/metadata/anvil/service/response.py", line 187, in _make_source
    self._special_fields.source_prefix: json_str(es_source['prefix']),
                                                 ~~~~~~~~~^^^^^^^^^^
KeyError: 'prefix'
2025-11-03 17:01:05,849    INFO Thread-4 (process_request_thread) azul.chalice: Returning 500 response with headers {"headers": {"Access-Control-Allow-Origin": "*", "Access-Control-Allow-Headers": "Authorization,Content-Type,X-Amz-Date,X-Amz-Security-Token,X-Api-Key", "Content-Security-Policy": "default-src 'self';img-src 'self' data:;script-src 'self';style-src 'self';frame-ancestors 'none';form-action 'self'", "Referrer-Policy": "strict-origin-when-cross-origin", "Strict-Transport-Security": "max-age=63072000; includeSubDomains; preload", "X-Content-Type-Options": "nosniff", "X-Frame-Options": "DENY", "X-XSS-Protection": "1; mode=block", "Cache-Control": "no-store"}}.
2025-11-03 17:01:05,849    INFO Thread-4 (process_request_thread) azul.chalice: … with a response body of length 133 being {"Code": "InternalServerError", "Message": "An internal server error occurred.", "RequestId": "4bbec8dc-6275-4f33-9a14-b07637c14c30"}
127.0.0.1 - - [03/Nov/2025 17:01:05] "GET /index/files?catalog=anvil11-it HTTP/1.1" 500 -
FAIL
…
======================================================================
FAIL: test_local_chalice_index_endpoints (integration_test.AzulChaliceLocalIntegrationTest.test_local_chalice_index_endpoints)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builds/ucsc/azul/test/integration_test.py", line 1806, in test_local_chalice_index_endpoints
    self.assertEqual(200, response.status_code, response.content)
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: 200 != 500 : b'{"Code":"InternalServerError","Message":"An internal server error occurred.","RequestId":"4bbec8dc-6275-4f33-9a14-b07637c14c30"}'

----------------------------------------------------------------------
Ran 16 tests in 513.596s

FAILED (failures=1)
make: *** [Makefile:259: integration_test] Error 1
```

--- a/test/integration_test.py
+++ b/test/integration_test.py
@@ -1855,7 +1855,7 @@ class AzulChaliceLocalIntegrationTest(AzulTestCase):
     catalog = first(config.integration_test_catalogs)
 
     def test_local_chalice_index_endpoints(self):
-        url = str(self.url.copy().set(path='index/files',
+        url = str(self.url.copy().set(path='repository/sources',
                                       query=dict(catalog=self.catalog)))
         response = requests.get(url)
         self.assertEqual(200, response.status_code, response.content)

--- a/test/integration_test.py
+++ b/test/integration_test.py
@@ -708,8 +708,8 @@ class IndexingIntegrationTest(IntegrationTestCase):
                             self._manifest_validators[format](catalog, response.data)
                             break
 
-                execution_ids = self._manifest_execution_ids(responses)
-                self.assertEqual(1, len(execution_ids))
+                    execution_ids = self._manifest_execution_ids(responses)
+                    self.assertEqual(1, len(execution_ids))
 
     def _manifest_execution_ids(self,
                                 responses: list[urllib3.HTTPResponse]

